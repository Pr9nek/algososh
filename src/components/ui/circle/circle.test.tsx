import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";
import { render, screen } from "@testing-library/react";

describe("Circle component snapshot tests", () => {

    // it("renders correctly without letter", () => {
    //     const tree = renderer.create(<Circle />).toJSON();
    //     expect(tree).toMatchSnapshot();
    // });

    it('рендер Circle без буквы', () => {
        render(<Circle letter="" />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle с буквами', () => {
        render(<Circle letter='A' />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle с head', () => {
        render(<Circle head='head' />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle с tail', () => {
        render(<Circle tail='tail' />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle с index', () => {
        render(<Circle index={0} />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle с пропом isSmall ===  true', () => {
        render(<Circle isSmall={true} />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle в состоянии default', () => {
        render(<Circle state={ElementStates.Default} />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle в состоянии changing', () => {
        render(<Circle state={ElementStates.Changing} />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

    it('рендер Circle в состоянии modified', () => {
        render(<Circle state={ElementStates.Modified} />);
        expect(screen.getByTestId('circle_container')).toMatchSnapshot();
    });

})