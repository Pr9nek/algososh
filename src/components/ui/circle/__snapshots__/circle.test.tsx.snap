// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component snapshot tests renders Circle in the changing state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle in the default state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle in the modified state 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle with head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle with index 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle with props isSmall ===  true 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="small"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle with tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Circle component snapshot tests renders Circle with Ð° letter 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders Circle without a letter 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders with react-element in head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle component snapshot tests renders with react-element in tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content "
      data-cy="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;
